### PR TITLE
Changing validation approach

### PR DESCRIPTION
**Pull request by dlatyshev**
Delete isOpened() methods from all classes. Rename Page to Base. Create general method to validate if a page was opened using a constructor.

### CODE REVIEW

File: src/main/java/io/customertimes/automation/pages/Base.java [hunk 1]
```diff
@@ -0,0 +1,20 @@
+package io.customertimes.automation.pages;
+
```
Comment (TyshchenkoRoma) [inline src/main/java/io/customertimes/automation/pages/Base.java:2]: del empty line

File: src/main/java/io/customertimes/automation/pages/AppLauncherBase.java [hunk 2]
```diff
@@ -1,18 +1,14 @@
 package io.customertimes.automation.pages;
 
 import com.codeborne.selenide.SelenideElement;
-import org.openqa.selenium.By;
-
 import static com.codeborne.selenide.Selenide.$;
 
-public class AppLauncherPage extends Page {
+public class AppLauncherBase extends Base {
+    private boolean isOpened = false;
```
Comment (TyshchenkoRoma) [inline src/main/java/io/customertimes/automation/pages/AppLauncherBase.java]: delete it.

File: src/main/java/io/customertimes/automation/pages/Base.java [hunk 3]
```diff
@@ -0,0 +1,20 @@
...
+
+import com.codeborne.selenide.Condition;
+import com.codeborne.selenide.SelenideElement;
+
+public abstract class Base {
+    protected boolean isOpened;
+
+    protected boolean pageIsOpened(SelenideElement... elements){
```
Comment (TyshchenkoRoma) [inline src/main/java/io/customertimes/automation/pages/Base.java:10]: why "..."

File: src/main/java/io/customertimes/automation/pages/Case.java [hunk 4]
```diff
@@ -1,18 +1,18 @@
...
-    @Override
-    public boolean pageIsOpened() {
-       return caseDetails.waitUntil(Condition.exist, 10000).exists();
+
+    public Case() {
+        this.isOpened = (pageIsOpened(caseDetails));
     }
 
 
+    public boolean isOpened() {
```
Comment (TyshchenkoRoma) [inline src/main/java/io/customertimes/automation/pages/Case.java]: we need this method?

File: src/main/java/io/customertimes/automation/pages/LoginBase.java [hunk 5]
```diff
@@ -2,14 +2,11 @@
 
 import com.codeborne.selenide.Condition;
 import com.codeborne.selenide.SelenideElement;
-import org.openqa.selenium.By;
-
 import static com.codeborne.selenide.Selenide.$;
 
-public class LoginPage extends Page {
+public class LoginBase extends Base {
```
Comment (TyshchenkoRoma) [inline src/main/java/io/customertimes/automation/pages/LoginBase.java]: LoginBase -> LoginPage

File: src/main/java/io/customertimes/automation/pages/SourceDocumentBase.java [hunk 6]
```diff
@@ -6,17 +6,15 @@
 import static com.codeborne.selenide.Selenide.$;
 import static com.codeborne.selenide.Selenide.$x;
 
-public class SourceDocumentPage extends Page {
+public class SourceDocumentBase extends Base {
```
Comment (TyshchenkoRoma) [inline src/main/java/io/customertimes/automation/pages/SourceDocumentBase.java]: SourceDocumentBase - > SourceDocumentPage

File: src/main/java/io/customertimes/automation/pages/SourceDocumentBase.java [hunk 7]
```diff
@@ -6,17 +6,15 @@
...
     private SelenideElement mergeBtn = $("button[name = \"merge\"]");
 
-    @Override
-    public boolean pageIsOpened() {
-        boolean splitButtonExists = splitBtn.waitUntil(Condition.appears, 10000).exists();
-        boolean mergeButtonExists = mergeBtn.waitUntil(Condition.appears, 10000).exists();
-        return splitButtonExists && mergeButtonExists;
+
+    public SourceDocumentBase() {
+        this.isOpened = pageIsOpened(splitBtn, mergeBtn, readyForAutomaticCaseProcessingBtn, enterCaseInformationManuallyBtn);
```
Comment (TyshchenkoRoma) [inline src/main/java/io/customertimes/automation/pages/SourceDocumentBase.java]: check title

File: src/test/java/io/customertimes/automation/steps/CaseSteps.java [hunk 8]
```diff
@@ -11,7 +10,7 @@
 
     @Then("^case details are opened$")
     public void caseDetailsAreOpened()  {
-        Assert.assertTrue(aCase.pageIsOpened());
+        Assert.assertTrue(aCase.isOpened());
```
Comment (TyshchenkoRoma) [inline src/test/java/io/customertimes/automation/steps/CaseSteps.java]: aCase - > Case

File: src/test/java/io/customertimes/automation/steps/LogInSteps.java [hunk 9]
```diff
@@ -1,25 +1,20 @@
...
+import io.customertimes.automation.pages.LoginBase;
 import io.customertimes.automation.utilities.TestUtils;
-import org.testng.Assert;
 
 import static com.codeborne.selenide.Selenide.*;
 
 
 public class LogInSteps extends TestUtils {
-    private LoginPage loginPage = new LoginPage();
+    private LoginBase loginPage = new LoginBase();
```
Comment (TyshchenkoRoma) [inline src/test/java/io/customertimes/automation/steps/LogInSteps.java]: LoginBase - LoginPage

File: src/test/java/io/customertimes/automation/steps/SourceDocumentSteps.java [hunk 10]
```diff
@@ -1,17 +1,16 @@
...
 import org.testng.Assert;
 
 public class SourceDocumentSteps {
-    SourceDocumentPage sourceDocument = new SourceDocumentPage();
+    SourceDocumentBase sourceDocument = new SourceDocumentBase();
 
     @Then("^details and file preview are opened$")
     public void detailsAndFilePreviewAreOpened() {
-        Assert.assertTrue(sourceDocument.pageIsOpened());
+        Assert.assertTrue(sourceDocument.isOpened());
```
Comment (TyshchenkoRoma) [inline src/test/java/io/customertimes/automation/steps/SourceDocumentSteps.java:14]: IsOpened -> pageIsOpened

File: src/main/java/io/customertimes/automation/pages/Case.java (hunk 4 above)
Comment (TyshchenkoRoma) [reply src/main/java/io/customertimes/automation/pages/Case.java]: is displayed

**Review by TyshchenkoRoma — APPROVED** [review]

File: src/main/java/io/customertimes/automation/pages/Base.java (hunk 1 above)
Comment (dlatyshev) [reply src/main/java/io/customertimes/automation/pages/Base.java:2]: Done.

**Pull request merged**